openedx/core/djangoapps/catalog/utils.py

```diff
@@ -315,48 +315,20 @@ def get_fulfillable_course_runs_for_entitlement(entitlement, course_runs):
     """
     Takes a list of course runs and returns only the course runs, sorted by start date, that:
 
-    1) Are currently running or in the future
+    1) Are currently running or start in the future
     2) A user can enroll in
-    3) A user can upgrade in
-    4) Are published
-    5) Are not enrolled in already for an active session
+    3) A user can upgrade to the entitlement mode
+    4) Are not already currently enrolled in a session
```

Comment:
nit: 'The user is not already enrolled in.

Also, maybe we can change this function from 'get_fulfillable_course_runs_for_entitlement' to 'get_fulfillable_sessions_for_entitlement' to keep with the language moving from course run to session